spidermon/contrib/scrapy/monitors.py

```diff
@@ -174,6 +174,99 @@ def test_verify_item_validation_error(self):
         self.assertTrue(item_validation_errors <= errors_threshold, msg=msg)
 
 
+@monitors.name("Field Coverage Monitor")
+class FieldCoverageMonitor(BaseScrapyMonitor):
+    """Validate if field coverage rules are met.
+
+    To use this monitor you need to enable the ``SPIDERMON_ADD_FIELD_COVERAGE``
+    setting, which will add information about field coverage to your spider
+    statistics.
+
+    To define your field coverage rules create a dictionary containing the
+    expected coverage for each field you want to monitor.
+
+    As an example, if the items you are returning from your spider are Python dictionaries
+    with the following format:
+
+    .. code-block:: python
+
+        {
+            "field_1": "some_value",
+            "field_2": "some_value",
+            "field_3": {
+                "field_3_1": "some_value",
+                "field_3_2": "some_value",
+            }
+        }
+
+    A set of rules may be defined as follows:
+
+    .. code-block:: python
+
+        # project/settings.py
+        SPIDERMON_FIELD_COVERAGE_RULES = {
+            "dict/field_1": 0.4,  # Expected 40% coverage for field_1
+            "dict/field_2": 1.0,  # Expected 100% coverage for field_2
+            "dict/field_3": 1.0,  # Expected 100% coverage for parent field_3
+            "dict/field_3/field_3_1": 0.5,  # Expected 50% coverage for nested field_3_1
```

Comment:
Maybe we should explicitly indicate in the documentation below whether the value for nested fields is relative to the parent field or absolute.

Reply:
Documentation was improved.